lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -130,6 +131,17 @@ def install_template
       @template_installed.make_true
     end
 
+    def discover_cluster_uuid
```

Comment:
a good way to remove the amount of indentation in a method is to return early for known situations. We can also move the catch into the method definition as it's pretty small. in this case meaning we can have instead:

```ruby
def discover_cluster_uuid
  return unless if defined?(plugin_metadata)
  cluster_info = client.get('/')
  plugin_metadata.set(:cluster_uuid, cluster_info['cluster_uuid'])
rescue => e
  @logger.error("Unable to retrieve elasticsearch cluster uuid", error => e.message)
  nil
end
```

Reply:
The problem with this is that even with the changes I have made to the DLQ tests, emitting *any* logging message at this point will break that entire series of tests, which is why it was `nil` to begin with. Certainly, the DLQ tests should be further addressed to prevent this problem but I'm not sure if we want to do that in this PR or a follow-up. Thoughts?

Reply:
ok let's comment out the logger entry and leave a `# TODO introducing this logging message breaks many tests that need refactoring`